### PR TITLE
Add a guide on how to use the seccomp gadget

### PR DESCRIPTION
**Pull request by marga-kinvolk**
I've added a guide based on the demo that I created for eBPF day. It includes all examples as files and goes through them step by step.

Fixes: #266

### CODE REVIEW

File: docs/guides/seccomp.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+```
+
+Once the pod is up, we can once again generate some traffic, like before.
+
+```
+$ kubectl get svc hello-python-service -n seccomp-demo
+NAME                   TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
+hello-python-service   LoadBalancer   10.111.82.101   <pending>     6000:31863/TCP   72s
+$ curl 10.111.82.101:6000
+Hello World!
```
Comment (alban) [inline docs/guides/seccomp.md:158]: ditto with kubectl port-forward

File: docs/guides/seccomp.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+**Remember that you need to set the right node name**.
+
+We can apply this modified file, which will change the existing resource
+(keeping the already gathered data) and then call `generate` again to get
+the profile generated:
+
+```
+$ kubectl apply -f docs/examples/seccomp/trace-external.yaml
+trace.gadget.kinvolk.io/seccomp configured
```
Comment (alban) [inline docs/guides/seccomp.md:212]: I wonder if that's something we should advertise...

In my PR "[seccomp: generate SeccompProfile for terminated pods](https://github.com/kinvolk/inspektor-gadget/pull/272)", I generate Seccomp Profiles directly upon pod termination, and no longer only after annotating with "generate". In that case, the "OutputMode" setting is read when reading the "start" operation, and changing it afterwards will not change the setting for those terminated pods.

**Review by alban — COMMENTED** [review]
Some comments but I like this guide!

Later, we could have a troubleshooting section, mentioning the things that could go wrong:
- If the annotation does not do anything, check if the node field is set correctly
- If some syscalls are missing, check the AllowPrivilegeEscalation
- Anything else?

File: docs/guides/seccomp.md (hunk 2 above)
Comment (marga-kinvolk) [reply docs/guides/seccomp.md:212]: Right, I didn't expect this to work myself, and discovered it just by chance when preparing the demo. It makes sense not to advertise it if we don't want to stick to this behavior.

**Comment by alban** [comment]
Btw when we have a cluster of more than 1 node, this guide is difficult to follow since we don't know in advance where the pod will be deployed. But this can be fixed in a future PR.

File: docs/guides/seccomp.md [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+Once we have captured the syscalls, we can ask the gadget to generate the
+corresponding profile, by setting the operation to `generate`.
+
+```
+$ kubectl annotate -n gadget trace/seccomp gadget.kinvolk.io/operation=generate
+trace.gadget.kinvolk.io/seccomp annotated
+```
+
+As the `outputMode` was set to `Status`, the generated policy will be
+stored in our CRD's status. We can look at it and check out the list of
```
[the remark this answers is not in the document]
Comment (marga-kinvolk) [reply docs/guides/seccomp.md]: Done

File: docs/guides/seccomp.md (hunk 1 above)
Comment (marga-kinvolk) [reply docs/guides/seccomp.md:168]: Done

File: docs/guides/seccomp.md (hunk 2 above)
Comment (marga-kinvolk) [reply docs/guides/seccomp.md]: I've redone this part without mentioning that calling generate directly would work. This means re-doing all the previous steps, which is a bit annoying, but I guess it's fine to just re-iterate what's going on.

**Comment by marga-kinvolk** [comment]
I've reworked the guide with the given comments and included a Troubleshooting section. Could you please take another look? Thanks!

File: docs/guides/seccomp.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+### Troubleshooting
+
+1. If the annotations don't do anything, check that the node field is set
+   correctly. You can also look at the `Status` field of the `Trace` for
+   other possible errors.
+
+2. If applying the policy causes the pod not to start, check that you're
+   setting `AllowPrivilegeEscalation=false` in the container's security
+   context, as having privilege escalation enabled doesn't work well with
+   seccomp. See #267 for more information.
```
Comment (alban) [inline docs/guides/seccomp.md:335]:
```suggestion
   seccomp. See [#267](https://github.com/kinvolk/inspektor-gadget/issues/267) for more information.
```

**Review by mauriciovasquezbernal — APPROVED** [review]
I tested it out and it works fine!

It's a very nice guide, I think we should take it as an example and other guides should have this quality.

**Review by alban — APPROVED** [review]
1 suggested change but LGTM!

File: docs/guides/seccomp.md [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+$ kubectl delete -f docs/examples/seccomp/unconfined.yaml
+pod "hello-python" deleted
+$ kubectl annotate -n gadget trace/seccomp gadget.kinvolk.io/operation=stop
+trace.gadget.kinvolk.io/seccomp annotated
+
+# Start a new trace and create the pod again
+$ kubectl annotate -n gadget trace/seccomp gadget.kinvolk.io/operation=start
+trace.gadget.kinvolk.io/seccomp annotated
+$ kubectl apply -f docs/examples/seccomp/trace-external.yaml
+trace.gadget.kinvolk.io/seccomp unchanged
```
[the remark this answers is not in the document]
Comment (marga-kinvolk) [reply docs/guides/seccomp.md:237]: True, this was a copy and paste error, I'll remove it. Thanks for catching it.

File: docs/guides/seccomp.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (marga-kinvolk) [reply docs/guides/seccomp.md]: Done

Comment (marga-kinvolk) [reply docs/guides/seccomp.md]: Done :)

**Comment by marga-kinvolk** [comment]
Comments applied. Thanks for the kind words :).

**Pull request merged**